### DISCUSSION TITLE
Error out with classed condition

### DISCUSSION THREAD

**Discussion by alexverse** [ropensci/targets#997] [Help] [answered]
Using `rlang::abort()` we can create classed conditions using the `class` param that can be caught or handled selectively, allowing for finer-grained error handling. This can be useful for Rmarkdown reports that execute pipelines and shiny apps that control targets workflows.

Is there a way to error out with classed conditions from targets pipeline? for example something like:

```
library(targets)
list(
  tar_target(
    name = abort,
    command = rlang::abort("FOO", class = "foo_error")
  )
)
```

and when calling `tar_make()`:

```
tryCatch({
  targets::tar_make()
}, foo_error = function(e) {
  rlang::inform("foo handled")
})
```

**Comment by wlandau** [comment, marked as answer]
That sounds reasonable. See 381d9678554fc4a13d994e9ceadd59a55b657298 for an implementation.

**Reply by alexverse** [reply]
Great! Changes in main are OK for custom error classes!
Thank you 🎄

**Reply by alexverse** [reply]
With these changes we can also use a custom error handling with base R:

```
library(targets)
list(
  tar_target(
    name = abort,
    command = stop(errorCondition("foo", class = "foo_error"))
  )
)
```

and then:

```
tryCatch({
  targets::tar_make()
}, foo_error = function(e) {
  message("foo handled")
})
````
